projects/tomcat/build.sh

```diff
@@ -19,10 +19,12 @@ cp -r "/usr/lib/jvm/java-17-openjdk-amd64/" "$JAVA_HOME"
 
 $ANT
 
-cp "output/build/lib/tomcat-coyote.jar" "$OUT/tomcat-coyote.jar"
-cp "output/build/lib/tomcat-util.jar" "$OUT/tomcat-util.jar"
+cd $SRC/tomcat/output/classes && jar cfv classes.jar . && mv ./classes.jar $OUT && cd $SRC/tomcat
 
-ALL_JARS="tomcat-coyote.jar tomcat-util.jar"
+LDAP_VERSION=$(curl -s 'https://api.github.com/repos/pingidentity/ldapsdk/tags?per_page=1' | jq -r .[].name)
+wget -O "$OUT/unboundid-ldapsdk.jar" "https://repo1.maven.org/maven2/com/unboundid/unboundid-ldapsdk/$LDAP_VERSION/unboundid-ldapsdk-$LDAP_VERSION.jar"
```

Comment:
Relying on wget for dependencies is not ideal. `unboundid` should be somewhere in the dependencies of Tomcat already if you can try to get it from there. I had a similar issue with spring-security (with gradle tho): https://github.com/CodeIntelligenceTesting/oss-fuzz/commit/7e9e1ed9a965d0068b906d530ec3cf43adddfe96

Reply:
If I can't find it in Tomcat can I build ldapsdk from source?

Reply:
I found it here https://github.com/apache/tomcat/blob/1086b2ef46a337e8c01feecce3a195e58bacffdb/build.xml#L240
I think it should be included in the jar if you add `<pathelement location="${unboundid.jar}"/>` to the `<path id="compile.classpath">` array. Let me know if works

Reply:
Thank you for the information! It helps me a lot :D

For now I added this two parts in my build.sh and I got the unboundid.jar now when building tomcat, please let me know if this modification is fine!

1. `sed -i '/<path id="compile.classpath">/a <pathelement location="${unboundid.jar}"/>' $SRC/tomcat/build.xml`

2. `sed -i '/depends="setup-bnd">/a <antcall target="downloadfile">'\
'<param name="sourcefile" value="${unboundid.loc}"/>'\
'<param name="destfile" value="${unboundid.jar}"/>'\
'<param name="destdir" value="${unboundid.home}"/>'\
'<param name="checksum.enabled" value="${unboundid.checksum.enabled}"/>'\
'<param name="checksum.algorithm" value="${unboundid.checksum.algorithm}"/>'\
'<param name="checksum.value" value="${unboundid.checksum.value}"/>'\
'</antcall>' $SRC/tomcat/build.xml`

Reply:
Glad I could help. It can work, but I recommend using `git patch` instead. it's a bit safer a easier to read. You can find an example here: https://github.com/CodeIntelligenceTesting/oss-fuzz/commit/402d392cd8b39dde35dbdd205b6bb0df713a51c8 

To get the diff, just edit the file, run `git diff | cat` and paste the result in the script

Reply:
Thank you for the tips! But for some reasons, when I try to cat input into patch.diff I encounter bad substitution error, I found out the reason is that cat somehow can't accept lines starting with "+", have you encountered this problem before? Maybe I will just use multiple line echo to the patch.diff

Reply:
I solved this problem with using echo and enclosing everything between ' ', now the problem is solved. But I still confuse why I cannot use cat to create a multi line file when there's a line starting with "+", maybe it's a bug.

Reply:
I think I did something wrong